tests/tests/swfs/avm2/decode_uri/Test.as

```diff
@@ -0,0 +1,56 @@
+﻿package {
+
+	public class Test {
+	}
+}
+
+import flash.utils.getDefinitionByName;
+
+var fns = ["decodeURI", "decodeURIComponent"];
+for each (var fnName in fns) {
+	var fn = getDefinitionByName(fnName);
+	trace("// " + fnName + "()");
+	trace(fn());
+	trace("");
+
+	trace("// " + fnName + "(undefined)");
+	trace(fn(undefined));
+	trace("");
+
+	trace("// typeof(" + fnName + "(undefined))");
+	trace(typeof(fn(undefined)));
+	trace("");
+
+	trace("// " + fnName + "(null)");
+	trace(fn(null));
+	trace("");
+
+	var input = "test";
+	trace("// " + fnName + "(\"" + input + "\")");
+	trace(fn(input));
+	trace("");
+
+	var input = "%3A";
+	trace("// " + fnName + "(\"" + input + "\")");
+	trace(fn(input));
+	trace("");
+	
+	var input = "%E0%A4%A";
+	trace("// " + fnName + "(\"" + input + "\")");
+	try {
+		trace(fn(input));
+	} catch (e) {
+		trace(e);
+	}
+	trace("");
+
+	var input = "\x05";
+	trace("// " + fnName + "(\"\\x05\")");
+	trace(fn(input));
+	trace("");
+
+	var input = "😭";
+	trace("// " + fnName + "(\"" + input + "\")");
+	trace(fn(input));
+	trace("");	
```

Comment:
I would suggest adding one with input being `encodeURI("😭")`. I'm not sure if decoding that works in Flash, though.

Reply:
It supports it in flash but not in my implementation, as i have not supported 4 bytes utf-8 yet

Reply:
Fixed now